javascript/test/config-file-tests.js

```diff
@@ -0,0 +1,165 @@
+const deleteFolderWithFiles = function (dir) {
+    if (fs.existsSync(dir)) {
+        fs.readdirSync(dir).forEach((file, index) => {
+            fs.unlinkSync(path.join(dir, file));
+        });
+        fs.rmdirSync(dir);
+    }
+};
+
+// Find your module using the --minesweeper_home variable and the default name 'minesweeper'
+const minesweeperHome = process.env.npm_config_minesweeper_home.endsWith("/")
+    ? process.env.npm_config_minesweeper_home
+    : process.env.npm_config_minesweeper_home + "/";
+const minesweeper = require(minesweeperHome + "minesweeper");
+
+describe('Config File Tests', function () {
+    describe('Runs with 2 exit code. Tag: Assignment1, Assignment2, Assignment3, Assignment4, Assignment5', function () {
+
+        var tempDir;
+        var minesweeperProcess;
+
+        beforeEach("Create temporary folder for file creation", function () {
+            // mkdtempSync requires the trailing `path.sep` to ensure a directory is created
+            tempDir = fs.mkdtempSync(os.tmpdir() + path.sep);
+            // tempDir does not have the trailing 'path.sep`
+            // TODO use path.join([...paths])
+            tempDir += path.sep;
+        });
+
+        afterEach("Remove temporary folder for file creation", function () {
+            deleteFolderWithFiles(tempDir);
+        });
+
+        it('Minesweeper exit because of invalid extension', function () {
+            // Creates the simple.cfg file
+            const lines = ['..*', '.,.', '...'];
+            fs.writeFileSync(tempDir + "simple", lines.join('\n') + '\n')
+
+            minesweeperProcess = spawnSync('node', [minesweeperHome + "minesweeper.js", tempDir + "simple"], { timeout: 3000 })
+            assert.strictEqual(minesweeperProcess.status, 2, "Wrong exit code for invalid board configuration file extension")
+        });
+
+        it('Minesweeper exit because of invalid characters (,)', function () {
+            // Creates the simple.cfg file
+            const lines = ['..*', '.,.', '...'];
+            fs.writeFileSync(tempDir + "simple.cfg", lines.join('\n') + '\n')
+
+            minesweeperProcess = spawnSync('node', [minesweeperHome + "minesweeper.js", tempDir + "simple.cfg"], { timeout: 3000 })
+            assert.strictEqual(minesweeperProcess.status, 2, "Wrong exit code for invalid (wrong characters) board configuration file")
+        });
+
+        it('Minesweeper exit because of new line character in last line missing', function () {
+            // Creates the simple.cfg file
+            const lines = ['..*', '...', '...'];
+            fs.writeFileSync(tempDir + "simple.cfg", lines.join('\n'))
+
+            minesweeperProcess = spawnSync('node', [minesweeperHome + "minesweeper.js", tempDir + "simple.cfg"], { timeout: 3000 })
+            assert.strictEqual(minesweeperProcess.status, 2, "Wrong exit code for invalid (new line missing) board configuration file")
+        });
```

Comment:
I am not entirely sure about this one...

Reply:
I managed to make my code passing all the tests except this one. What I am concerned about is the fact that the last new lines should not matter that much, so I will change this test from "checking whether the file content is valid or not" to check if the GUI produced by minesweeper contains the right number of columns/rows.

In order terms:

`['..*', '...', '...'].join('\n') + '\n'` and  `['..*', '...', '...'].join('\n')` should both produce a 3 x 3 board

@lechlm What do you think?

Reply:
That's okay from my side, but then for this test case the first option (['..*', '...', '...'].join('\n') + '\n') should be used because in my program I check for this last new line character since it needs to be there according to the program description. So the second option (['..*', '...', '...'].join('\n')), which I used in this version of the test case, produce an exit code 2 (see program description).

Reply:
@lechlm My bad! The specifications of the program clearly say that all the lines must be terminated with '\n' so your test is valid.